analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -450,6 +463,425 @@ public void Finally_NestedFinally()
                 "InInnerFinally",
                 "AfterInnerFinally",
                 "AfterOuterFinally");
+
+            validator.TagStates("InOuterFinally").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("InInnerFinally").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("InInnerTry").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("AfterInnerFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+
+            validator.TagStates("AfterOuterFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInFinally"";
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",          // With Exception thrown by Tag("InTry")
+                "InCatch",          // With Exception thrown by `throw`
+                "AfterCatch",
+                "AfterCatch");      // Should go away after https://github.com/SonarSource/sonar-dotnet/pull/5745 is done.
+
+            validator.TagStates("InCatch").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasUnknownException(x))
+                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
```

Comment:
This assertion repeats many times too, with a different pair. `validator.ValidateTagHasExceptionAndUnknownException("InCatch")`.

Or `validator.ValidateTagHasSystemExceptionAndUnknownException("InCatch")`